streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/FlinkCluster.java

```diff
@@ -138,23 +144,34 @@ public URI getActiveAddress() {
         return null;
     }
 
-    public boolean verifyConnection() {
-        if (address == null) {
-            return false;
-        }
-        String[] array = address.split(",");
-        for (String url : array) {
-            try {
-                new URI(url);
-            } catch (Exception ignored) {
+    public boolean verifyClusterConnection() {
+        if (ExecutionMode.REMOTE.equals(this.getExecutionModeEnum()) ||
+            ExecutionMode.YARN_SESSION.equals(this.getExecutionModeEnum())) {
+            if (address == null) {
                 return false;
             }
-            try {
-                HttpClientUtils.httpGetRequest(url, RequestConfig.custom().setConnectTimeout(2000).build());
-                return true;
-            } catch (Exception ignored) {
-                //
+            String[] array = address.split(",");
+            for (String url : array) {
+                try {
+                    new URI(url);
+                } catch (Exception ignored) {
+                    return false;
+                }
+                try {
+                    String restUrl;
+                    if (ExecutionMode.REMOTE.equals(this.getExecutionModeEnum())) {
+                        restUrl = url + "/overview";
+                    } else {
+                        restUrl = url + "/proxy/" + this.clusterId + "/overview";
+                    }
+                    String result = HttpClientUtils.httpGetRequest(restUrl, RequestConfig.custom().setConnectTimeout(2000).build());
+                    JacksonUtils.read(result, Overview.class);
+                    return true;
+                } catch (Exception ignored) {
+                    //
+                }
             }
+            return false;
         }
         return false;
```

Comment:
Can 3 `return false` be simplified to one?

Reply:
> 

The first `try..catch` is to ensure that a group of addresses, each is a legitimate url, if any of them is an invalid url then return,
The second `try...catch` is to find an active and connected address in a group of addresses

Reply:
From your reply, I think they can be simplified . I still don't understand why don't simplify them.

Reply:
> From your reply, I think they can be simplified . I still don't understand why don't simplify them.

   e.g: address is : http://192.168.0.1,xxyy:a.b.c,192.168.1.2

  ```
    String[] array = address.split(",");
            for (String url : array) {
                try {
                    new URI(url);
                } catch (Exception ignored) {
                    return false;
                }
                try {
                    String restUrl;
                    if (ExecutionMode.REMOTE.equals(this.getExecutionModeEnum())) {
                        restUrl = url + "/overview";
                    } else {
                        restUrl = url + "/proxy/" + this.clusterId + "/overview";
                    }
                    String result = HttpClientUtils.httpGetRequest(restUrl, RequestConfig.custom().setConnectTimeout(2000).build());
                    JacksonUtils.read(result, Overview.class);
                    return true;
                } catch (Exception ignored) {
                    //
                }
            }
```

then parse `xxyy:a.b.c` first try..catch will return, second will find only one if connection is ok and return.

Reply:
> > From your reply, I think they can be simplified . I still don't understand why don't simplify them.
> 
> e.g: address is : http://192.168.0.1,xxyy:a.b.c,192.168.1.2
> 
> ```
>   String[] array = address.split(",");
>           for (String url : array) {
>               try {
>                   new URI(url);
>               } catch (Exception ignored) {
>                   return false;
>               }
>               try {
>                   String restUrl;
>                   if (ExecutionMode.REMOTE.equals(this.getExecutionModeEnum())) {
>                       restUrl = url + "/overview";
>                   } else {
>                       restUrl = url + "/proxy/" + this.clusterId + "/overview";
>                   }
>                   String result = HttpClientUtils.httpGetRequest(restUrl, RequestConfig.custom().setConnectTimeout(2000).build());
>                   JacksonUtils.read(result, Overview.class);
>                   return true;
>               } catch (Exception ignored) {
>                   //
>               }
>           }
> ```
> 
> then parse `xxyy:a.b.c` first try..catch will return, second will find only one if connection is ok and return.

first try...catch Will check whether each url is ok, but the second is not

Reply:
After read this code 3 times, I think I understand.

The first `try catch` will return directly when any url isn't legal, right?  For example, the url isn't legal, the next url won't be check.

If right, I think the code should be refactored. It's very very hard to read, I read it too many times to understand it. 😭

How about this? The `isLegalUrl` is an util method.
```
for (String url : array) {
  if(!isLegalUrl(url)) {
    return false;
  }
  // try to connect yarn ...
}
```

If It's still wrong, please correct me.